### PR TITLE
Upgradeconfig collector

### PR DESCRIPTION
**Pull request by sedroche**
### What type of PR is this?
cleanup

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_ https://issues.redhat.com/browse/OSD-5542 & https://issues.redhat.com/browse/OSD-4339

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [x] Ran `make generate` command locally to validate code changes
- [x] Included documentation changes with PR

### CODE REVIEW

**Comment by mrbarge** [comment]
Tested this okay and it worked; I'll give the code a :eyes: next.

File: pkg/controller/machineconfigpool/machineconfigpool_controller.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	if instance.Name != "worker" {
```
[the remark this answers is not in the document]
Comment (sedroche) [reply pkg/controller/machineconfigpool/machineconfigpool_controller.go:87]: Yeah, I think that makes sense. I'll add it in

File: pkg/metrics/collector.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cv "github.com/openshift/managed-upgrade-operator/pkg/clusterversion"
+	ucm "github.com/openshift/managed-upgrade-operator/pkg/upgradeconfigmanager"
+)
+
+var (
```
Comment (dofinn) [inline pkg/metrics/collector.go:13]: Please add metrics tag to each metric. 
tag ref: https://github.com/openshift/managed-upgrade-operator/blob/361eebdd01f2bf57099e6e776242d04a87277ea0/pkg/metrics/metrics.go#L21
implemenation ref: https://github.com/openshift/managed-upgrade-operator/blob/361eebdd01f2bf57099e6e776242d04a87277ea0/pkg/metrics/metrics.go#L105-L109

**Comment by dofinn** [comment]
\lgtms after the metricsTag have been added.

**Comment by mrbarge** [comment]
I'm also fine for this to go through, will let @dofinn make the call when he's finished his next tests.

**Comment by dofinn** [comment]
@sedroche validating workercompletion time in UpgradeConfig and and CVO history target versions.

**Comment by dofinn** [comment]
Testing now doing sequential upgrades and monitoring prometheus.

**Pull request merged**